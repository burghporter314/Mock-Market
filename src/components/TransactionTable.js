import React, { Component } from 'react';
import { Table, Nav } from 'react-bootstrap';

class TransactionTable extends Component {
    constructor(props) {
        super(props);
        this.state = {
            headers:["", "", "", ""],
            rows: [
                {
                    title: "",
                    bought: "",
                    current: "",
                    amount: "",
                },
                {
                    title: "",
                    bought: "",
                    current: "",
                    amount: "",
                }
            ],
        }
    }

    render() {
        return(
            <Table striped bordered hover>
                <thead>
                    <tr>
                        {(this.props.headers ? this.props.headers: this.state.headers).map((header, index) => (
                            <th key={index}>{header}</th>
                        ))}
                    </tr>
                </thead>
                <tbody>
                    {(this.props.rows ? this.props.rows : this.state.rows).map((row, index) => (
                        <tr key={index} onClick={this.props.processRowClick(row[0])}>
                            {Object.keys(row).map((key, index) => (
                                <td key={index}>{row[key]}</td>
                            ))}
                        </tr>
                    ))}
                </tbody>
            </Table>
        );
    }
}

export default TransactionTable;