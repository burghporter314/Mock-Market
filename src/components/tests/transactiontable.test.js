import React from 'react';
import TransactionTable from '../TransactionTable';
import renderer from 'react-test-renderer';

it('should match snapshot with empty headers and rows', () => {
    const component = renderer.create(
        <TransactionTable
        headers={["", "", "", ""]}
        rows={[
            {
                title: "",
                bought: "",
                current: "",
                amount: "",
            },
            {
                title: "",
                bought: "",
                current: "",
                amount: "",
            }]
        }/>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
})

it('should match snapshot with empty headers and populated rows', () => {
    const component = renderer.create(
        <TransactionTable
        headers={["", "", "", ""]}
        rows={[
            {
                title: "title",
                bought: "bought",
                current: "current",
                amount: "amount",
            },
            {
                title: "title",
                bought: "bought",
                current: "current",
                amount: "amount",
            }]
        }/>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
})

it('should match snapshot with populated headers and empty rows', () => {
    const component = renderer.create(
        <TransactionTable
        headers={["header1", "header2", "header3", "header4"]}
        rows={[
            {
                title: "title",
                bought: "bought",
                current: "current",
                amount: "amount",
            },
            {
                title: "title",
                bought: "bought",
                current: "current",
                amount: "amount",
            }]
        }/>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
})

it('should match snapshot with populated headers and partial rows', () => {
    const component = renderer.create(
        <TransactionTable
        headers={["header1", "header2", "header3", "header4"]}
        rows={[
            {
                title: "title",
                bought: "bought",
                current: "current",
            },
            {
                title: "title",
                bought: "bought",
                current: "current",
            }]
        }/>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
})

it('should match snapshot with partial headers and populated rows', () => {
    const component = renderer.create(
        <TransactionTable
        headers={["header1", "header2", "header3"]}
        rows={[
            {
                title: "title",
                bought: "bought",
                current: "current",
                amount: "amount",
            },
            {
                title: "title",
                bought: "bought",
                current: "current",
                amount: "amount",
            }]
        }/>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
})