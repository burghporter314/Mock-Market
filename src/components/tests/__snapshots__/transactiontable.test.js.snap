// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot with empty headers and populated rows 1`] = `
<table
  className="table table-striped table-bordered table-hover"
>
  <thead>
    <tr>
      <th>
        
      </th>
      <th>
        
      </th>
      <th>
        
      </th>
      <th>
        
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        title
      </td>
      <td>
        bought
      </td>
      <td>
        current
      </td>
      <td>
        amount
      </td>
    </tr>
    <tr>
      <td>
        title
      </td>
      <td>
        bought
      </td>
      <td>
        current
      </td>
      <td>
        amount
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`should match snapshot with empty headers and rows 1`] = `
<table
  className="table table-striped table-bordered table-hover"
>
  <thead>
    <tr>
      <th>
        
      </th>
      <th>
        
      </th>
      <th>
        
      </th>
      <th>
        
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        
      </td>
      <td>
        
      </td>
      <td>
        
      </td>
      <td>
        
      </td>
    </tr>
    <tr>
      <td>
        
      </td>
      <td>
        
      </td>
      <td>
        
      </td>
      <td>
        
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`should match snapshot with partial headers and populated rows 1`] = `
<table
  className="table table-striped table-bordered table-hover"
>
  <thead>
    <tr>
      <th>
        header1
      </th>
      <th>
        header2
      </th>
      <th>
        header3
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        title
      </td>
      <td>
        bought
      </td>
      <td>
        current
      </td>
      <td>
        amount
      </td>
    </tr>
    <tr>
      <td>
        title
      </td>
      <td>
        bought
      </td>
      <td>
        current
      </td>
      <td>
        amount
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`should match snapshot with populated headers and empty rows 1`] = `
<table
  className="table table-striped table-bordered table-hover"
>
  <thead>
    <tr>
      <th>
        header1
      </th>
      <th>
        header2
      </th>
      <th>
        header3
      </th>
      <th>
        header4
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        title
      </td>
      <td>
        bought
      </td>
      <td>
        current
      </td>
      <td>
        amount
      </td>
    </tr>
    <tr>
      <td>
        title
      </td>
      <td>
        bought
      </td>
      <td>
        current
      </td>
      <td>
        amount
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`should match snapshot with populated headers and partial rows 1`] = `
<table
  className="table table-striped table-bordered table-hover"
>
  <thead>
    <tr>
      <th>
        header1
      </th>
      <th>
        header2
      </th>
      <th>
        header3
      </th>
      <th>
        header4
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        title
      </td>
      <td>
        bought
      </td>
      <td>
        current
      </td>
    </tr>
    <tr>
      <td>
        title
      </td>
      <td>
        bought
      </td>
      <td>
        current
      </td>
    </tr>
  </tbody>
</table>
`;
